server/server.py

```
@@ -0,0 +1,335 @@
+        """Handle requests for resource aliases directing them to the appropriate local resource."""
+        use_version = version if version else self._default_version
+        use_resource = f'{resource}{extension}' if resource else f'{self._default_resource}{extension}'
+        effective_path = f'/{use_version}/{use_resource}' if use_version else f'/{use_resource}'
+        if req.path == effective_path:
+            _logger.warning(f"Requested path '{req.path}' is not actually an alias - check your routing rules!")
+            OntologyResource().on_get(req=req, resp=resp, version=use_version, resource=use_resource)
+        elif not use_resource.startswith(self._required_prefix):
+            _logger.warning(f"Requested path '{req.path}' is not a valid ontology resource alias")
+            raise falcon.HTTPNotFound(description=f"Invalid ontology resource '{req.path}'")
+        else:
+            _logger.warning(f"Requested path '{req.path}' is an alias for '{effective_path}'. Redirecting.")
+            raise falcon.HTTPSeeOther(location=f'{PATH_PREFIX}{effective_path}')
+
+class RemoteResource:
+    """Falcon Request Handler that deals with redirects to remote resources of any type."""
+
+    def __init__(self, remote_target: str | None = None):
+        """Set per-handler-instance defaults for missing local resource path parts."""
+        self._remote_target = remote_target
+        self._target_is_url = validators.url(self._remote_target)
+
+    def on_get(self, req: falcon.Request, resp: falcon.Response) -> None:
+        """Handle requests for remote resource aliases."""
+        if self._target_is_url:
+            raise falcon.HTTPMovedPermanently(location=self._remote_target)
+        else:
+            _logger.error(f"Unable to redirect to configured remote target '{self._remote_target}': not a valid URL")
+            raise falcon.HTTPRouteNotFound()
```

These routes are hard coded at the start of the app, and the validation can be done then. An invalid RemoteResource is an error that the server can not handle gracefully - the server is misconfigured and it should announce itself as that so it can be fixed. As it is currently written an incorrectly set FAVICON_URL will silently fail, only announcing itself in the logs when someone access it, while the error is detectable as soon as the application is deployed and is only fixable by a sysadmin fixing the configuration.

I suggest modifying the `RemoteResource__init__()` to raise a ValueError if the remote resource is not a valid URL. A incorrectly configured application will then fail to boot, the sysadmin can immediately investigate, see the error message, and fix the issue. Better to never deploy an incorrectly configured application in the first place!

---

Fixed in recent commits